### PR TITLE
feat(suite): settings desktop update info based on current update state

### LINKED ISSUE #4463: Update ready for installation in Settings contextual improvement
Design is OK, we just should differentiate two states:
1. When an update is available to download and install (this state is covered well):
image
2. When the update has already been updated, but not installed well:
image

A quick copy improvement could help here.
When the update has been downloaded, but not installed yet, we could show copy:

```
Update ready for installation // EDIT this line is omitted, approved by b-irving
You are currently running version [current version nr.]
Version [downloaded (updated) version nr.] has been downloaded and is ready for installation.
```

### PR DESCRIPTION
**Pull request by tomasklim**
Close #4463

### CODE REVIEW

File: packages/suite/src/views/settings/index.tsx [hunk 1]
```diff
@@ -382,12 +394,14 @@ const Settings = () => {
                                         ),
                                     }}
                                 />
-                                {!['checking', 'not-available'].includes(desktopUpdate.state) &&
+                                {(desktopUpdate.state === 'available' ||
+                                    desktopUpdate.state === 'downloading' ||
+                                    desktopUpdate.state === 'ready') &&
```
Comment (matejkriz) [inline packages/suite/src/views/settings/index.tsx]: It's definitely matter of taste, feel free to ignore this comment, but do you really find multiple `||` more readable than: 
```suggestion
                                {[ 'available', 'downloading', 'ready'].includes( desktopUpdate.state) &&
```
?

Comment (matejkriz) [reply packages/suite/src/views/settings/index.tsx]: ok, it's because of the typescript...

**Review by matejkriz — APPROVED** [review]

File: packages/suite/src/views/settings/index.tsx [hunk 2]
```diff
@@ -35,6 +35,18 @@ const buildCurrencyOption = (currency: string) => ({
     label: currency.toUpperCase(),
 });
 
+const getUpdateStateMessage = (state: 'available' | 'downloading' | 'ready') => {
```
Comment (matejkriz) [inline packages/suite/src/views/settings/index.tsx]: consider defining `desktopUpdate.state` type as enum

Comment (tomasklim) [reply packages/suite/src/views/settings/index.tsx]: f2f22d6f677f9271303b90c06f07424843be4165

File: packages/suite/src/reducers/suite/desktopUpdateReducer.ts [hunk 3]
```diff
@@ -10,6 +10,17 @@ import { UpdateInfo, UpdateProgress, UpdateWindow } from '@suite-types/desktop';
  * - not-available: Currently on the latest version
  * - downloading: Currently downloading the latest version
  * - ready: Latest version is downloaded and going to be installed on the next restart
+ */
+export enum UpdateState {
+    Checking = 'checking',
```
Comment (matejkriz) [inline packages/suite/src/reducers/suite/desktopUpdateReducer.ts:15]: nit: We have ~7 enums defined so far, and most of them use `UPPER_SNAKE_CASE` for keys. I personally don't like it, but let's keep it consistent.  (I would prefer camelCase for the keys and basically for all constants and PascalCase just for names of enums, interfaces, React components and so on...)

Comment (tomasklim) [reply packages/suite/src/reducers/suite/desktopUpdateReducer.ts:15]: I did it according to TS docs https://www.typescriptlang.org/docs/handbook/enums.html and the way most people in `Medium` articles do it. I agree with you that also a lot of people uses PascalCase for names and UPPER_CASE for keys. However, you do not like it, so I would stick to the TS docs style.

We have a mix of everything in the codebase (even a mix in one enum). 
```
export enum Flags {
    None = 0,
    hasEmail = 1 << 0,
    hasBookmark = 1 << 1
}
```

To sum it up, I would leave the way I did it. We have only a few enums in the codebase and I think we should use them more in the future. Before we start using it, we can have a discussion and unite all the existing enums to the style we decide.

Comment (matejkriz) [reply packages/suite/src/reducers/suite/desktopUpdateReducer.ts:15]: Ok, I just created a low-hanging-fruit issue for it: https://github.com/trezor/trezor-suite/issues/4504

**Pull request merged**